### PR TITLE
Sortable images on media multiple

### PR DESCRIPTION
**Pull request by motchju**
Hi,

The title should be self explanatory.

As written in file-link-multiple.blade.php, you need to change another configuration file in asgard.core.core.php

Only one "bug" remaining, not directly related to this pull request.
When you add one/multiple image, the sortable functionality doesn't work because of the html added, not inside <figure> tags

Is there a reason for that? That will be most likely my next task.

### CODE REVIEW

File: Resources/views/admin/fields/file-link-multiple.blade.php [hunk 1]
```diff
@@ -110,5 +131,26 @@
...
             });
         });
+
+        $(".jsThumbnailImageWrapper").sortable({
+            tolerance: 'pointer',
+            revert: 'invalid',
+            forceHelperSize: true,
+            update:function(event, ui) {
+                var dataSortable = $(this).sortable('toArray', {attribute: 'data-id'});
+                console.log(dataSortable);
```
Comment (nWidart) [inline Resources/views/admin/fields/file-link-multiple.blade.php]: You left a console.log here.

File: Resources/views/admin/fields/file-link-multiple.blade.php [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+{!! Theme::style('vendor/jquery-ui/jquery-ui.min.js') !!}
+
+The only way for me was to modify config -> asgard.core.core.php
+in admin-assets add
+
+'jquery-ui' => ['module' => 'dashboard:vendor/jquery-ui/jquery-ui.min.js']
+
+and load it in admin-required-assets -> js
+
+'jquery-ui',
```
Comment (nWidart) [inline Resources/views/admin/fields/file-link-multiple.blade.php]: This is how you should load it: https://github.com/AsgardCms/Dashboard/blob/master/Http/Controllers/Admin/DashboardController.php#L105

**Comment by nWidart** [comment]
I don't understand what you mean with 

> Only one "bug" remaining, not directly related to this pull request.
When you add one/multiple image, the sortable functionality doesn't work because of the html added, not inside

Can't you change the template html to make it work ?

> Is there a reason for that? That will be most likely my next task.

Reason for what ?

**Comment by motchju** [comment]
If you check the html, after a clean refresh, the html is the following, inside jsThumbnailImageWrapper:

```
<figure data-id="25" class="ui-sortable-handle">
   <img alt="" src="/assets/media/aspirateur-brico-sf-002_mediumThumb.jpg">
   <a data-id="25" href="#" class="jsRemoveLink">
       <i class="fa fa-times-circle removeIcon"></i>
   </a>
</figure>
<figure data-id="26" class="ui-sortable-handle">
   <img alt="" src="/assets/media/aspirateur-brico-sf-002_mediumThumb.jpg">
   <a data-id="26" href="#" class="jsRemoveLink">
       <i class="fa fa-times-circle removeIcon"></i>
   </a>
</figure>
```
If you add one new image (done via ajax of course, you do not reload the page, you will have

```
<figure data-id="25" class="ui-sortable-handle">
   <img alt="" src="/assets/media/aspirateur-brico-sf-002_mediumThumb.jpg">
   <a data-id="25" href="#" class="jsRemoveLink">
       <i class="fa fa-times-circle removeIcon"></i>
   </a>
</figure>
<figure data-id="26" class="ui-sortable-handle">
   <img alt="" src="/assets/media/aspirateur-brico-sf-002_mediumThumb.jpg">
   <a data-id="26" href="#" class="jsRemoveLink">
       <i class="fa fa-times-circle removeIcon"></i>
   </a>
</figure>
<!--- New image --->
<img alt="" src="/assets/media/site-sprite_mediumThumb.png">
<a data-id="19" href="#" class="jsRemoveSimpleLink"><i class="fa fa-times-circle removeIcon"></i></a>
```
It's not inside a figure tags.

**Comment by nWidart** [comment]
that can be fixed though, no ?

**Comment by motchju** [comment]
yes can be I will check if I can add the item to be wrapped inside figure tags, or maybe easier change the sortable to react on img not on figure.

I will try to pass every step tomorrow

**Comment by motchju** [comment]
"Issue" fixed concerning figure tags, in fact I was using both:  the single image and the multiple image in the same page, so the code was duplicated.

In file-link-blade.php the insert was not wrapped into the figure tags, but it's the case in file-link-multiple.blade.php

Anyway I change it. 

Now about how to include the js dependency, if I am not wrong:

https://asgardcms.com/en/docs/media-module/relations

```php
@include('media::admin.fields.file-link-multiple', [
    'entityClass' => 'Modules\\\\Product\\\\Domain\\\\Entities\\\\Product',
    'entityId' => $product->id,
    'zone' => 'gallery'
])
```

So yes, I can do it via the controllers where I use the include, but for me the include should be autonomous, no need to add dependency in every controller that use the include.

If it's still not clear or if I am unclear, I am available via skype if you want.

**Comment by motchju** [comment]
For the scrutinizer error, please check if your composer is up to date:

https://laracasts.com/discuss/channels/testing/travis-build-errors-because-of-composer-dependency

https://github.com/Intervention/image/issues/464

**Comment by nWidart** [comment]
the asset needs to be linked with the partial via a view composer. Of course I wasn't talking about controller. What I linked was only an example for you to see how it has to be done.

**Comment by motchju** [comment]
Hi,

So now the JS are loaded via the MediaServiceProvider.
Let me know if it's fine, still new to Laravel , it's working however not sure it's "done correctly".

**Comment by mikemand** [comment]
You'll need to rebase on the current master. This is reverting some of my changes from #27.

**Comment by motchju** [comment]
I did that yesterday, I will double check tomorrow, I saw some difference in file-link-multiple.blade.php

**Comment by nWidart** [comment]
Thanks!
I see you used a view creator, but in my opinion a view composer would be best.

**Comment by motchju** [comment]
Update done

**Comment by nWidart** [comment]
Thank you ! I'll test this as soon as possible.

**Comment by mikemand** [comment]
Thank you for fixing #27 compatibility, but you're still not up-to-date with Master. e699e68ebe2808b0cd4baa06a55cb1b85fe99762 renamed the variable from `$thumbnailSize` to `$thumbnail`.

**Comment by motchju** [comment]
Fix $thumbnailSize to $thumbnail. Thank you for the follow up

**Comment by mikemand** [comment]
Thank you @motchju! :+1:

**Comment by nWidart** [comment]
Pulled your branch locally , migrated the media module, and getting this: https://cloudup.com/c2lcIRJifS4

**Comment by nWidart** [comment]
Found the issue, you used blade template for javascript. Blade is not a javascript templating engine. :)

Line 52 in `file-link-multiple.blade.php`
``` php
var html = '<figure data-id="{{ data.result.imageableId }}"><img src="' + data.result.path + '" alt=""/>' +
// TO
var html = '<figure data-id="' + data.result.imageableId +'"><img src="' + data.result.path + '" alt=""/>' +
```

**Comment by nWidart** [comment]
You're also not loading the jquery ui correctly: 

https://cloudup.com/cp9kTLZVBLC

**Comment by motchju** [comment]
corrected for the Js issue, currently looking into the jquery ui

**Comment by nWidart** [comment]
You need to replace:

``` php
view()->composer('media::admin.fields.file-link-multiple', function () {
    $this->assetPipeline = app('Modules\Core\Foundation\Asset\Pipeline\AssetPipeline');
    $this->assetManager = app('Modules\Core\Foundation\Asset\Manager\AssetManager');

    $this->assetManager->addAsset('jquery-ui-draggable.js', 'dashboard:vendor/jquery-ui/ui/minified/draggable.min.js');

    $this->assetPipeline->requireJs('jquery-ui-core.js');
    $this->assetPipeline->requireJs('jquery-ui-draggable.js');

});
```
With:
``` php
view()->composer('media::admin.fields.file-link-multiple', function () {
    $assetPipeline = app('Modules\Core\Foundation\Asset\Pipeline\AssetPipeline');
    $assetManager = app('Modules\Core\Foundation\Asset\Manager\AssetManager');

    $assetManager->addAsset('jquery-ui-draggable.js', Module::asset('dashboard:vendor/jquery-ui/ui/minified/draggable.min.js'));

    $assetPipeline->requireJs('jquery-ui-core.js');
    $assetPipeline->requireJs('jquery-ui-draggable.js');
});
```

But even that doesn't work so not sure what you did.

**Comment by nWidart** [comment]
Remove the view composer stuff completely and add this on top of the file-link-multiple view:

```
<script src="{{ Module::asset('dashboard:vendor/jquery-ui/jquery-ui.min.js') }}"></script>
```

**Comment by nWidart** [comment]
Also other issue with the drag/drop animation: http://quick.as/nB5UVd0g

Once that is fixed, create a new pull request with a clean commit history please as this one is messed up with a lot of odd commit messages.
You can squash commits and rename commit messages for this.

**Pull request closed without merging**